src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -87,17 +89,19 @@ private BoundExpression VisitCompoundAssignmentOperator(BoundCompoundAssignmentO
 
                 // transformedLHS = storeNonEvent + loweredRight
                 rewrittenAssignment = rewriteAssignment(lhsRead);
+                Debug.Assert(rewrittenAssignment.Type is { });
 
                 // Final conditional
                 var condition = _factory.Conditional(isEvent, invokeEventAccessor.ToExpression(), rewrittenAssignment, rewrittenAssignment.Type);
 
-                rewrittenAssignment = new BoundSequence(node.Syntax, eventTemps.ToImmutableAndFree(), sequence.ToImmutableAndFree(), condition, condition.Type);
+                rewrittenAssignment = new BoundSequence(node.Syntax, eventTemps.ToImmutableAndFree(), sequence.ToImmutableAndFree(), condition, condition.Type!);
```

Comment:
>condition.Type!); [](start = 144, length = 17)

How do we know?

Reply:
Because we passed a non-null type when we created the bound node through the factory on the previous line.

---
In reply to: [386623764](https://github.com/dotnet/roslyn/pull/42068#discussion_r386623764) [](ancestors = 386623764)